task/slack-webhook-notification/0.1/slack-webhook-notification.yaml

```
@@ -41,12 +49,140 @@ spec:
+        }
+
+        function dumpFile {
+          filePath=$1
+
+          read -r -d '' ret << EOM
+        *${filePath}:*
+        \`\`\`
+        $(cat "${filePath}")
+        \`\`\`
+        EOM
+
+          echo "${ret}"
+        }
+
+        function dumpSubmodule {
+          name=$1
+
+          path=$(git config -f .gitmodules --get submodule."${name}".path)
+          url=$(git config -f .gitmodules --get submodule."${name}".url)
+
+          current_commit=$(git -C "${path}" rev-parse HEAD)
+          previous_commit=$(git diff HEAD~1 "${path}" | grep commit | head -n 1 | awk '{print $3;}')
+          if [ "${previous_commit}" = "" ]; then
+            previous_commit=${current_commit}
+          fi
+
+          commits=None
+          if [ "${previous_commit}" != "${current_commit}" ]; then
+            commits=$(git -C "${path}" log --pretty=format:'%h %<(27,trunc)%an %s' --abbrev-commit "${previous_commit}".."${current_commit}")
+          fi
+
+          read -r -d '' ret << EOM
+        *Submodule ${name} (<${url}/commits/${current_commit}|commits>):*
+        \`\`\`
+        ${commits}
+        \`\`\`
+        EOM
+
+          echo "${ret}"
+        }
+
+
+        # --------------
+        #  ARGS PARSING
+        # --------------
+
+        echo "Parsing $@"
+
+        FILES=""
+        SUBMODULES=""
+
+        while [[ $# -gt 0 ]]; do
+          case $1 in
+          --files)
+            shift
+            while [[ $# -gt 0 ]] && ! [[ "$1" =~ --.* ]]; do
+              FILES+=" $1"
+              shift
+            done
+            ;;
+          --submodules)
+            shift
+            while [[ $# -gt 0 ]] && ! [[ "$1" =~ --.* ]]; do
+              SUBMODULES+=" $1"
+              shift
+            done
+            ;;
+          esac
+        done
+
+
+        # -------------------
+        #  PARAMS VALIDATION
+        # -------------------
+
         if [ -f "/etc/secrets/$KEY_NAME" ]; then
           WEBHOOK_URL=$(cat "/etc/secrets/$KEY_NAME")
         else
           echo "Secret not defined properly"
           exit 1
         fi
-        curl -X POST -H 'Content-type: application/json' --data "{\"text\":\"$MESSAGE\"}" $WEBHOOK_URL
+
+
+        # ------
+        #  DUMP
+        # ------
+
+        slack_message=${MESSAGE}
+
+        if [ "$FILES" != "" ]; then
+          slack_message+="\n$(dumpSeparator)"
+        fi
+
+        for file in ${FILES}; do
+          content=$(dumpFile "${file}")
+          slack_message+="\n${content}"
+        done
+
+        if [ "$SUBMODULES" != "" ]; then
+          slack_message+="\n$(dumpSeparator)"
+        fi
+
+        for submodule in ${SUBMODULES}; do
+          content=$(dumpSubmodule "${submodule}")
+          slack_message+="\n${content}"
+        done
+
+        escaped_slack_message=$(escapeDoubleQuote "${slack_message}")
+
+        curl -X POST -H 'Content-type: application/json' --data "{\"text\":\"${escaped_slack_message}\"}" "$WEBHOOK_URL"
```

Are double quotes the only character that needs escaping? It would be safer to use `jq` (but the `image` would need to have `jq` installed, you could use e.g. `quay.io/konflux-ci/appstudio-utils:ab6b0b8e40e440158e7288c73aff1cf83a2cc8a9@sha256:24179f0efd06c65d16868c2d7eb82573cce8e43533de6cea14fec3b7446e0b14`)
```suggestion
        curl -X POST -H 'Content-type: application/json' --data "$(jq --compact-output --null-input --arg message "$slack_message" '{text: $message}')" "$WEBHOOK_URL"
```

---

Slack asks to escape (but not with \) "&<>" 

- it's fine if they are inside a ``` ``` block
- So the MESSAGE (from the user) remains. I think it's better to let the user fix that if he wants to. Otherwise we might escape too much (for example what is inside a ``` ``` clock cerated by the user

I changed to your proposal